.changeset/flat-shirts-shake.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Skips trailing slash handling for internal paths
```

Comment:
This might need some rewording for the end-users; I doubt the users know anything about "internal paths".